api/envoy/extensions/filters/http/rate_limit_quota/v3/rate_limit_quota.proto

```diff
@@ -0,0 +1,159 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.rate_limit_quota.v3;
+
+import "envoy/config/core/v3/base.proto";
+import "envoy/config/core/v3/config_source.proto";
+import "envoy/config/core/v3/extension.proto";
+import "envoy/type/v3/ratelimit_strategy.proto";
+
+import "google/protobuf/duration.proto";
+import "google/rpc/status.proto";
+
+import "xds/annotations/v3/status.proto";
+import "xds/type/matcher/v3/matcher.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.rate_limit_quota.v3";
+option java_outer_classname = "RateLimitQuotaProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/rate_limit_quota/v3;rate_limit_quotav3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#not-implemented-hide:]
+// [#protodoc-title: Rate Limit Quota]
+// Rate limit :ref:`configuration overview <config_http_filters_rate_limit_quota>`.
+// [#extension: envoy.filters.http.rate_limit_quota]
+
+message RateLimitQuota {
+  // [#comment:TODO(sergiitk): preview mode matchers]
+
+  // Configuration for the gRPC service that hosts an external rate limit
+  // provider. If not specified, any calls to the rate limit service will
+  // immediately return success.
+  config.core.v3.ConfigSource rlqs_server = 1;
+
+  // The application domain to use when calling the service.
+  string domain = 2;
+
+  // Unified matchers.
+  xds.type.matcher.v3.Matcher bucket_matchers = 3;
+}
+
+// [#next-free-field: 7]
+message RateLimitQuotaBucketSettings {
```

Comment:
I assume that this is expected to be the action type that plugs into the matcher tree?  If so, please state that in a comment.

Reply:
Addressed in https://github.com/envoyproxy/envoy/pull/19793/commits/d6a23241f24983b997e043bf0bbd032f5821230f.